openapi.yaml

```diff
@@ -0,0 +1,473 @@
+      operationId: "allPost"
+      parameters:
+      - name: "sort"
+        in: "query"
+        description: "Sort the list of posts by property"
+        required: false
+        type: "string"
+        enum:
+          - "new"
+          - "best"
+          - "daily"
+      - name: "limit"
+        in: "query"
+        description: "Limit number of posts return from server"
+        required: false
+        type: "string"
+      - name: "page"
+        in: "query"
+        description: "How many rows to skip"
+        required: false
+        type: "string"
+      - name: "query"
+        in: "query"
+        description: "Keywords to search"
+        required: false
+        type: "string"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            type: "array"
+            items:
+              $ref: "#/definitions/Post"
+        default:
+          description: "error operation"
+          schema:
+            $ref: "#/definitions/ApiErrorResponse"
+  /posts/{id}/{action}:
+    put:
+      tags:
+      - "post"
+      summary: "Actions for the post"
+      operationId: "actionPost"
+      parameters:
+      - name: "id"
+        in: "path"
+        description: "Post id to view/save/share"
+        required: true
+        type: "string"
+      - name: "action"
+        in: "path"
+        description: "Post id to view/save/share"
+        required: true
+        type: "string"
+        enum:
+          - "save"
+          - "share"
+          - "view"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Post"
+        default:
+          description: "error operation"
+          schema:
+            $ref: "#/definitions/ApiErrorResponse"
+      security:
+      - UserSecurity: []
+    delete:
+      tags:
+      - "post"
+      summary: "Remove action for the post"
+      operationId: "RemoveActionPost"
+      parameters:
+      - name: "id"
+        in: "path"
+        description: "Post id"
+        required: true
+        type: "string"
+      - name: "action"
+        in: "path"
+        description: "Post id"
+        required: true
+        type: "string"
+        enum:
+          - "save"
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Post"
+        default:
+          description: "error operation"
+          schema:
+            $ref: "#/definitions/ApiErrorResponse"
+      security:
+      - UserSecurity: []
+
+securityDefinitions:
+  UserSecurity:
+    type: basic
+
+definitions:
+  User:
+    type: "object"
+    required:
+    - "email"
+    properties:
+      id:
+        type: "string"
+        description: "Id for the user"
+      email:
+        type: "string"
+        description: "Email for the user"
+      token:
+        type: "string"
+        description: "Token for the user when login by email"
```

Comment:
this should not be included here. this is not the database schema so do not define all properties here. just define what the client needs.